nautobot_ssot_ipfabric/diffsync/adapter_nautobot.py

```diff
@@ -122,7 +122,9 @@ def load_device(self, filtered_devices: List, location):
                 diffsync=self,
                 name=device_record.name,
                 model=str(device_record.device_type),
-                role=str(device_record.device_role) if str(device_record.device_role) else DEFAULT_DEVICE_ROLE,
+                role=str(device_record.device_role.cf.get("ipfabric_type"))
+                if str(device_record.device_role.cf.get("ipfabric_type"))
+                else str(device_record.device_role),
```

Comment:
why we don't use the `DEFAULT_DEVICE_ROLE` here?

Reply:
This is the Nautobot adapter so we want to load whatever role is actually applied to the Device object whether it is an actual value or blank. We use DEFAULT_DEVICE_ROLE in the IP Fabric adapter in the scenario that the device in IP Fabric doesn't have a role ("devType").